### PR TITLE
fix:cleanup entrypoint.sh for UDI

### PR DESCRIPTION
**Pull request by svor**
Signed-off-by: Valerii Svydenko <vsvydenk@redhat.com>

It's not possible to execute commands with **sudo** in entrypoint.sh. Can we proceed without removing unused _/etc/alternatives/java_ and _/etc/alternatives/javac_?

Fix for https://issues.redhat.com/browse/CRW-2862

### CODE REVIEW

**Comment by nickboldt** [comment]
My concern with this change (and the reason I added the alternatives switch) is that in https://issues.redhat.com/browse/CRW-2591 we saw a problem where the debugger and debuggee were not the same Java version.

What do you use in Che's UDI image to support multiple JDKs?

**Comment by svor** [comment]
It doesn't work as expected, I continue investigate the problem

File: devspaces-udi/etc/entrypoint.sh [hunk 1]
```diff
@@ -53,21 +53,13 @@ fi
...
 else
   if [ ! -z "${JAVA11_HOME}" ]; then
-    export JAVA_HOME=${JAVA11_HOME}
-    export PATH="${JAVA11_HOME}/bin:${PATH}"
-    sudo rm -f /etc/alternatives/java
-    sudo ln -s ${JAVA11_HOME}/bin/java /etc/alternatives/java
-    sudo rm -f /etc/alternatives/javac
-    sudo ln -s ${JAVA11_HOME}/bin/javac /etc/alternatives/javac
+    echo "JAVA_HOME=${JAVA11_HOME}" >> "${HOME}"/.bashrc
+    echo "PATH=${JAVA11_HOME}/bin:${PATH}" >> "${HOME}"/.bashrc
```
Comment (nickboldt) [inline devspaces-udi/etc/entrypoint.sh]: this should be an `export PATH...` in the .bashrc; same for `export JAVA_HOME=...`

for example in my ~/.bashrc I have:

```
export PATH=$PATH:~/bin
export JAVA_HOME=/usr/lib/jvm/java
export MAVEN_OPTS="-Xms512m -Xmx3000m"
export GITUSER=nickboldt
```

**Review by nickboldt — APPROVED** [review]
approved, with my small change. let's run it and see if it breaks java debugger/debuggee support :D

**Comment by svor** [comment]
By making **export** in entrypoint.sh - env was set only into its own environment. By adding ENVs into .bashrc we can solve the problem

**Comment by nickboldt** [comment]
> By making **export** in entrypoint.sh - env was set only into its own environment. By adding ENVs into .bashrc we can solve the problem

Agreed, but should the .bashrc simply define the vars, or EXPORT them to the shell, so they're more pervasive?

**Comment by svor** [comment]
> Agreed, but should the .bashrc simply define the vars, or EXPORT them to the shell, so they're more pervasive?

In case if user will need java8, it could change it by adding environment var in the devfile and then the right value of JAVA_HOME and PATH will be added into .bashrc. 
What do you mean by EXPORT them to the shell?

File: devspaces-udi/etc/entrypoint.sh (hunk 1 above)
Comment (svor) [reply devspaces-udi/etc/entrypoint.sh]: thanks

**Comment by nickboldt** [comment]
> What do you mean by EXPORT them to the shell?

I mean this:

https://github.com/redhat-developer/devspaces-images/pull/218/commits/1e65e2851480d12649270521571d705b2aa82be8

**Pull request merged**